plugin/mqtt/client.go

```diff
@@ -142,32 +149,42 @@ func (m *Client) ConnectionHandler(client paho.Client) {
 // Cleanup recursively removes a topic
 func (m *Client) Cleanup(topic string, retained bool) error {
 	statusTopic := topic + "/status"
-	if !m.Client.Subscribe(topic+"/#", m.Qos, func(c paho.Client, msg paho.Message) {
+	if !m.client.Subscribe(topic+"/#", m.Qos, func(c paho.Client, msg paho.Message) {
 		if len(msg.Payload()) == 0 || msg.Topic() == statusTopic {
 			return
 		}
 
 		m.log.TRACE.Printf("delete: %s", msg.Topic())
-		m.Client.Publish(msg.Topic(), m.Qos, true, []byte{})
+		m.Publish(msg.Topic(), true, "")
+
 	}).WaitTimeout(request.Timeout) {
 		return api.ErrTimeout
 	}
 
-	time.Sleep(time.Second)
+	time.Sleep(request.Timeout)
```

Comment:
This is a 10s delay during startup?

Reply:
added a dynamic timeout,
the time is still 1 second if the mqtt broker is fast, but can take longer if the broker is slow.

I also added a cleanup on shutdown to make the cleanup on startup faster.

Reply:
> will think about making it not hard coded but based on when the last message was reset.

Lovely solution!